components/fxa-client/ios/FxAClient/FirefoxAccount.swift

```diff
@@ -201,109 +146,243 @@ open class FirefoxAccount {
     open func checkAuthorizationStatus(completionHandler: @escaping (IntrospectInfo?, Error?) -> Void) {
-        queue.async {
+        DispatchQueue.global().async {
             do {
-                let infoBuffer = try FirefoxAccountError.unwrap { err in
-                    fxa_check_authorization_status(self.raw, err)
-                }
-                let msg = try! MsgTypes_IntrospectInfo(serializedData: Data(rustBuffer: infoBuffer))
-                fxa_bytebuffer_free(infoBuffer)
-                let tokenInfo = IntrospectInfo(msg: msg)
+                let tokenInfo = try self.checkAuthorizationStatusSync()
                 DispatchQueue.main.async { completionHandler(tokenInfo, nil) }
             } catch {
                 DispatchQueue.main.async { completionHandler(nil, error) }
             }
         }
     }
 
+    open func checkAuthorizationStatusSync() throws -> IntrospectInfo {
+        let ptr = try rustCall { err in
+            fxa_check_authorization_status(self.raw, err)
+        }
+        defer { fxa_bytebuffer_free(ptr) }
+        let msg = try! MsgTypes_IntrospectInfo(serializedData: Data(rustBuffer: ptr))
+        return IntrospectInfo(msg: msg)
+    }
+
     /// This method should be called when a request made with
     /// an OAuth token failed with an authentication error.
     /// It clears the internal cache of OAuth access tokens,
     /// so the caller can try to call `getAccessToken` or `getProfile`
     /// again.
     open func clearAccessTokenCache(completionHandler: @escaping (Void, Error?) -> Void) {
-        queue.async {
+        DispatchQueue.global().async {
             do {
-                try FirefoxAccountError.unwrap { err in
-                    fxa_clear_access_token_cache(self.raw, err)
-                }
+                try self.clearAccessTokenCacheSync()
                 DispatchQueue.main.async { completionHandler((), nil) }
             } catch {
                 DispatchQueue.main.async { completionHandler((), error) }
             }
         }
     }
 
+    open func clearAccessTokenCacheSync() throws {
+        try rustCall { err in
+            fxa_clear_access_token_cache(self.raw, err)
+        }
+    }
+
     /// Disconnect from the account and optionaly destroy our device record.
     /// `beginOAuthFlow(...)` will need to be called to reconnect.
     open func disconnect(completionHandler: @escaping (Void, Error?) -> Void) {
-        queue.async {
+        DispatchQueue.global().async {
             do {
-                try FirefoxAccountError.unwrap { err in
-                    fxa_disconnect(self.raw, err)
-                }
+                try self.disconnectSync()
                 DispatchQueue.main.async { completionHandler((), nil) }
-                self.tryPersistState()
             } catch {
                 DispatchQueue.main.async { completionHandler((), error) }
             }
         }
     }
+
+    open func disconnectSync() throws {
```

Comment:
As above, we might find our future selves wondering why there aren't async variants of these. But...there's a lot of them! I retract what I said earlier about maybe adding async variants for consistency, we should just document that the async ones are there for b/w compat.